xDrip/Scenes/Entries/EditTraining/EditTrainingInteractor.swift

```diff
@@ -14,20 +14,63 @@ import UIKit
 
 protocol EditTrainingBusinessLogic {
     func doLoad(request: EditTraining.Load.Request)
+    
+    func doCancel(request: EditTraining.Cancel.Request)
+    func doDone(request: EditTraining.Done.Request)
 }
 
 protocol EditTrainingDataStore {
-    
+    var mode: EditTraining.Mode? { get set }
 }
 
 final class EditTrainingInteractor: EditTrainingBusinessLogic, EditTrainingDataStore {
     var presenter: EditTrainingPresentationLogic?
     var router: EditTrainingRoutingLogic?
+    let formattingWorker = EditTrainingFormattingWorker()
+    var mode: EditTraining.Mode?
+    
+    var localEntry: EditTraining.LocalEntry?
```

Comment:
We should delegate more responsibility to the worker. The worker should handle transformations and saving. This would lead to removing this property.